doc/plugin_tutorial/tuto1/src/g_tuto1.mlg

```diff
@@ -145,12 +145,11 @@ END
   it gives an error message that is basically impossible to understand. *)
 
 VERNAC COMMAND EXTEND ExploreProof CLASSIFIED AS QUERY
-| ![ proof ] [ "Cmd9" ] ->
+| ![ proof_query ] [ "Cmd9" ] ->
```

Comment:
I have mixed feeling about the annotations, that seem to overlap with the classification for the stm.
Here there is already a `CLASSIFIED AS QUERY` annotation, adding another one `![proof_query]` seems unnecessary, or at least a call for having a single class of annotations. If the ones of the STM are not fine grained enough, we can improve them, for example `CLASSIFIED AS PROOF QUERY` could cover your use case.

Reply:
Note that it could also be an extra piece of the declaration, eg `... CLASSIFIED AS QUERY READS PSTATE`

Reply:
One big issue when trying to merge them is that classification is dynamic (the one in funind with `(* TASSI: n'importe quoi ! *)` is probably the worst but smaller scale dynamism is used in other places), so we can't use them to specify a static type.
Also `proof_query` isn't a great name as the command is free to do non-proof side effects, but I have no better idea.

Reply:
Yes, the STM classification can be dynamic to cover corner cases, but I'm talking more about the syntax rather than the type. The point is that the two concepts overlap, hence they should not grow orthogonally and organically IMO. The STM classification is about declaring what a command does but if we forget about the needs of the STM, one may also include what part of the system state it reads/write. The STM could even use that one day. The extra classification info can be used by coqpp to give a precise type to the code as well.

What I'm objecting is really that two related piece of metadata are written in different places and with a different syntax/scope.

Reply:
>Note that it could also be an extra piece of the declaration, eg ... CLASSIFIED AS QUERY READS PSTATE

We could do that but wouldn't we still need a more local version? like `=> { vtbla }` for classification.

>this one makes things quite worse

It looks about the same to me, except I guess for having to pass `is_ontop` just for a warning.

Reply:
IIRC the `CLASSIFY AS` affects the entire block.
One could split the block into smaller ones just to declare a different classification (I guess).

Reply:
> this one makes things quite worse

I meant: it adds many mo things like `![ proof ]` that was already somewhat related to the classification.

Reply:
I concur with @SkySkimmer `![ proof]` is only marginally related to the classification, as least for now. For example parsing state etc... is not reflected.